pkg/storage/expose/process_memory.go

```diff
@@ -0,0 +1,454 @@
+package expose
+
+import (
+	"encoding/binary"
+	"errors"
+	"fmt"
+	"io"
+	"os"
+	"strconv"
+	"strings"
+
+	"github.com/loopholelabs/silo/pkg/storage"
+)
+
+const PageSize = 4096
+const PageShift = 12
+
+const ReadBufferSize = 4 * 1024 * 1024 // 4mb should be fairly fast
+
+// pagemap flags
+const PagemapFlagSoftDirty = 1 << 55
+const PagemapFlagExclusive = 1 << 56
+const PagemapFlagFilepage = 1 << 61
+const PagemapFlagSwapped = 1 << 62
+const PagemapFlagPresent = 1 << 63
+const PagemapMaskPfn = (1 << 55) - 1
+const PagemapMaskSwapType = (1 << 5) - 1
+const PagemapSwapOffsetShift = 5
+const PagemapMaskSwapOffset = (1 << 50) - 1
+
+// kpageflags
+const KflagLocked = 1
+const KflagError = 1 << 1
+const KflagReferenced = 1 << 2
+const KflagUptodate = 1 << 3
+const KflagDirty = 1 << 4
+const KflagLru = 1 << 5
+const KflagActive = 1 << 6
+const KflagSlab = 1 << 7
+const KflagWriteback = 1 << 8
+const KflagReclaim = 1 << 9
+const KflagBuddy = 1 << 10
+const KflagMmap = 1 << 11
+const KflagAnon = 1 << 12
+const KflagSwapcache = 1 << 13
+const KflagSwapbacked = 1 << 14
+const KflagCompoundHead = 1 << 15
+const KflagCompoundTail = 1 << 16
+const KflagHuge = 1 << 17
+const KflagUnevictable = 1 << 18
+const KflagHwPoison = 1 << 19
+const KflagNopage = 1 << 20
+const KflagKSM = 1 << 21
+const KflagTHP = 1 << 22
+const KflagBalloon = 1 << 23
+const KflagZeroPage = 1 << 24
+const KflagIdle = 1 << 25
+
+/**
+ * ProcessMemory
+ *
+ */
+type ProcessMemory struct {
+	pid int
+}
+
+/**
+ * Smap
+ *
+ */
+type Smap struct {
```

Comment:
https://github.com/loopholelabs/smaps <- can we add this functionality into this lib?